src/app/(protected)/admin/animals/add/page.tsx

```diff
@@ -0,0 +1,185 @@
+"use client";
+
+import { SubmitHandler, useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Button } from "@/components/ui/button";
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import {
+  CreateAnimalInputDto,
+  createAnimalInputSchema,
+} from "@/app/use-cases/animals/create-animal/create-animal.dto";
+import { Loader2 } from "lucide-react";
+
+export default function AddAnimalPage() {
```

Comment:
Let's refactor this page route to `/admin/animals` only, as we usually have with backend routes.

Currently registering an animal in the backend route is POST `/api/animals`, for the add page on the frontend we can have `/admin/animals`.

For the edit animal page work (in another ticket), we can use [Next.js' dynamic routes](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes) to dynamic pull animal data to fill out the form.

The idea is that an animal's data could also be viewed/edited through the URL instead, for example:

Page `/admin/animals/41f53680-8fe0-4248-8745-6f4775379975` would render the page with the data for the animal under that id.